### ISSUE TITLE
Collapse attribution behind a control

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/Web-Map-Custom-Element#216]
There are usability issues with the attribution (some of which are mostly apparent on small screens/in small sized maps).

## Problems

Here's a default size map with added `opacity` to the attribution container to demonstrate:

1. Controls may be hidden behind the attribution (means we fail SC [2.4.7 Focus Visible](https://www.w3.org/TR/WCAG21/#focus-visible)):<br>
  ![attribution-z-index](https://user-images.githubusercontent.com/26493779/100800489-f6dcd000-3426-11eb-8ed7-47ded1d99b7a.png)<br>

2. There are no constraints on the element containing the attribution links, which may result in the following depending on the amount of layer sources and/or the map element's size:<br>
  ![attribution-overload](https://user-images.githubusercontent.com/26493779/98679427-4cf6b000-2360-11eb-95a6-002cc900c271.png)<br>

3. Sensory overload
   - Visual
   - Auditory (screen readers announce everything in the attribution on page load because it is visible)

4. Each attribution link adds to the amount of <kbd>Tab</kbd> key presses required to sequentially navigate past the map.

## Potential solution

Collapse attribution behind a control, like [MapBox](https://maps4html.org/HTML-Map-Element-UseCases-Requirements/examples/create-map.html#mapbox-embed) does:

![toggle-attribution](https://user-images.githubusercontent.com/26493779/143277052-1d83909f-10f0-442c-b62a-fe5db5de0a71.png)

Collapsing attribution may have several benefits:

- There's no visible Layout Shift (so does not contribute to the [Cumulative Layout Shift](https://web.dev/cls/)) when populating the attribution container with attributions after fetching layer sources.<br>
  <img src="https://user-images.githubusercontent.com/26493779/127612277-dc724fba-7d41-42b8-97ab-ece05b2350d7.png">

- Improves [Perceived Performance](https://developer.mozilla.org/en-US/docs/Learn/Performance/Perceived_performance). 

- Mitigates sensory overload for some users.

- Less <kbd>Tab</kbd> key presses required to sequentially navigate past the map.

**Comment by prushforth** [comment]
FWIW, the widget is supposed to de-dupe the attribution, so in the case of many layers from the same source, only one attribution entry should be created.  This may not work, but that's the theory.

**Comment by Malvoz** [comment]
> widget is supposed to de-dupe the attribution

That is probably the case, I simply modified the DOM in devtools to present the potential issue.

**Comment by prushforth** [comment]
How persnickety would Web Platform Tests be about the UI to the attribution control?  I like the approach [MapBox](https://maps4html.org/HTML-Map-Element-UseCases-Requirements/examples/create-map.html#mapbox-embed) takes, but some browsers might like to be more explicit.

**Comment by Malvoz** [comment]
@prushforth I don't know TBH, but hiding attribution behind controls may have several benefits.

There's no visible Layout Shift (so does not contribute to the [Cumulative Layout Shift](https://web.dev/cls/)) when populating the attribution container with attributions after fetching layer sources. It improves the [Perceived Performance](https://developer.mozilla.org/en-US/docs/Learn/Performance/Perceived_performance). Additionally there are (on initial load) less things to be announced by screen readers which helps with reducing sensory overload for some users.

Even though a native implementation would have good performance, I don't think there's a way around the potential layout shift (which is really apparent on slow connections) so it may even be that implementers would prefer hiding attribution behind a control.
